src/core/control/xojfile/XmlParserHelper.cpp

```diff
@@ -28,44 +28,39 @@
 #include "filesystem.h"
 
 
-XmlParserHelper::AttributeMap::AttributeMap(const char** attributeNames, const char** attributeValues) {
-    // Get array length, and verify that it is identical for names and values
-    std::size_t attribute_count = 0;
-    while (attributeNames[attribute_count] != nullptr) {
-        xoj_assert(attributeValues[attribute_count] != nullptr);
-        ++attribute_count;
-    }
-    xoj_assert(attributeValues[attribute_count] == nullptr);
-
-    // Allocate space for string views
-    this->names.resize(attribute_count);
-    this->values.resize(attribute_count);
-
-    // Reference strings
-    for (size_t i = 0; i < attribute_count; ++i) {
-        this->names[i] = xoj::util::utf8(attributeNames[i]).sv();
-        this->values[i] = xoj::util::utf8(attributeValues[i]).sv();
-    }
-}
-
-auto XmlParserHelper::AttributeMap::operator[](const std::u8string_view name) const
-        -> std::optional<std::u8string_view> {
-    auto it = std::ranges::find(this->names, name);
-    if (it != this->names.end()) {
-        // Name was found
-        return this->values[as_unsigned(std::distance(this->names.cbegin(), it))];
+XmlParserHelper::AttributeMap::AttributeMap(const char** attributeNames, const char** attributeValues):
+        names(attributeNames), values(attributeValues) {}
+
+auto XmlParserHelper::AttributeMap::operator[](std::u8string_view name) const -> std::optional<ViewType> {
+    for (auto it = this->names; *it != nullptr; ++it) {
+        if ((*it | xoj::util::utf8) == name) {
+            // Name was found
+            return reinterpret_cast<const char8_t*>(this->values[as_unsigned(std::distance(this->names, it))]) |
```

Comment:
So, the point of utf8_view is that casting to `const char8_t*` is undefined behaviour, so this line is a no go...